server/src/main/java/com/google/android/attestation/AuthorizationList.java

```diff
@@ -284,23 +284,33 @@ private static boolean findBooleanAuthorizationListEntry(
     return Optional.ofNullable(entry).map(ASN1OctetString::getOctets);
   }
 
-  private static Optional<UserAuthType> findOptionalUserAuthType(
+  private static Optional<Set<UserAuthType>> findOptionalUserAuthType(
       Map<Integer, ASN1Primitive> authorizationMap, int tag) {
     Optional<Long> userAuthType = findOptionalLongAuthorizationListEntry(authorizationMap, tag);
     return userAuthType.map(AuthorizationList::userAuthTypeToEnum);
   }
 
   // Visible for testing.
-  static UserAuthType userAuthTypeToEnum(long userAuthType) {
-    if (userAuthType == 0L) {
-      return USER_AUTH_TYPE_NONE;
-    } else if (userAuthType == 1L) {
-      return PASSWORD;
-    } else if (userAuthType == 2L) {
-      return FINGERPRINT;
-    } else if (userAuthType == UINT32_MAX) {
-      return USER_AUTH_TYPE_ANY;
+  static Set<UserAuthType> userAuthTypeToEnum(long userAuthType) {
+    Set<UserAuthType> result = new HashSet<>();
+    if (userAuthType == 0) {
+      result.add(USER_AUTH_TYPE_NONE);
```

Comment:
Minor comment: Return early here so the structure of the 'else' statement is not necessary:
if (userAuthType == 0) {
  return Set.of(USER_AUTH_TYPE_NONE);
}

Set<UserAuthType> result = new HashSet<>();
if ((userAuthType & 1L) == 1L) {
       result.add(PASSWORD);
} 
...
// then check if result is empty.

Reply:
Thanks for your review and suggestion. Addressed in c67d8aef21ecad80c4f83cc63da80534651c5c12.